src/main/java/org/truffleruby/core/rope/RopeCache.java

```diff
@@ -50,7 +50,7 @@ public Rope getRope(Rope string, CodeRange codeRange) {
     }
 
     @TruffleBoundary
-    public Rope getRope(String string) {
+    public Rope getCachedPath(String string) {
```

Comment:
I don't like this change. `RopeCache` is general purpose. If it happens to be only used on certain way, then maybe have a helper method somewhere else. Moreover, Java has a `Path` class and that's what I'd expect a method called `getCachedPath` to return -- not a `Rope`.

Reply:
Would renaming the method be enough? I want to convey this should not be called with user-generated strings though.
Maybe I should move this out in its own class.
General purpose often means it's incorrect or sub-optimal for at least some of the use-cases so I prefer to be fairly precise when all usages do the same thing.

Reply:
Apologies if I was unclear. I think the method name should remain and if you want a cache specific to paths, just create a new `RopeCache` instance. So calls would look something like `context.getPathCache().getRope(<j.l.String>)`.